clang/lib/Format/TokenAnnotator.cpp

```diff
@@ -2633,12 +2633,20 @@ class AnnotatingParser {
              PreviousNotConst->MatchingParen->Previous->isNot(tok::kw_template);
     }
 
-    if ((PreviousNotConst->is(tok::r_paren) &&
-         PreviousNotConst->is(TT_TypeDeclarationParen)) ||
-        PreviousNotConst->is(TT_AttributeRParen)) {
+    if (PreviousNotConst->is(tok::r_paren) &&
+        PreviousNotConst->is(TT_TypeDeclarationParen)) {
       return true;
     }
 
+    auto InTypeDecl = [&]() {
+      for (auto Next = Tok.Next; Next; Next = Next->Next)
```

Comment:
A more robust and efficient way would be to annotate the class name in `UnwrappedLineParser` as most of the work has already been done there. I'll submit a patch soon.

Reply:
See #124891.